official-antistasi-community/A3-Antistasi
# 🧭 Navigation Grid Node&Time Reduction

## PR Type
* Bug
* Change

### PRs to merge first:
Collections addition commit never happened 😉.

Some code is duplicated from base Antistasi for easy coping out of the mission. Therefore, the new code is under `/Tools/NavGridPP/functions/`

### What I have changed:
* ⏳ Navigation grid generation time has been slashed down.
* 📉 Nodes are simplified, reducing overall count.
* 📍 Remaining nodes are carefully chosen to be: In a junction or on turns.
* ↔ Nodes increase in density as another road approaches.

### And why:
* 🗺 Reducing count of nodes allows path finding to be quicker.
* 🚚 Increase in density as another road approaches allows `nearestTerrainObject` from a virtualised convoy to find the correct road (Not another parallel road). And vice-verse allow real convoys to find the correct navigation grid node.

### Please specify which Issue this PR Resolves.
closes https://github.com/official-antistasi-community/A3-Antistasi/issues/1693
closes https://github.com/official-antistasi-community/A3-Antistasi/issues/1692

### Please verify the following and ensure all checks are completed.

* [x] Have you loaded the mission in LAN host? (Arma 3 2.00)

### Is further testing or are further changes required?
1. [ ] No
* Yes

### How can the changes be tested?
Follow included quick start on any community map and ensure that there are no error
[Quick Start Guide](https://github.com/official-antistasi-community/A3-Antistasi/blob/CS-NavGridPP/Tools/NavGridPP/Please%20Read%20Me.md)
